{"uid":"ee91158a744e7d39","name":"User is able to login,this should fail","fullName":"tests.test_login.TestLogin#test_user_login","historyId":"156d8f497cfb2d6c2f80e1b63e77e862","time":{"start":1683293949010,"stop":1683293949010,"duration":0},"description":"User is able to login","descriptionHtml":"<p>User is able to login</p>\n","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55607d8e8fe3 <unknown>\n#1 0x55607d627d36 <unknown>\n#2 0x55607d664c4d <unknown>\n#3 0x55607d664d61 <unknown>\n#4 0x55607d6a06a4 <unknown>\n#5 0x55607d6858ed <unknown>\n#6 0x55607d69e232 <unknown>\n#7 0x55607d685693 <unknown>\n#8 0x55607d65803a <unknown>\n#9 0x55607d65917e <unknown>\n#10 0x55607d8aadbd <unknown>\n#11 0x55607d8aec6c <unknown>\n#12 0x55607d8b84b0 <unknown>\n#13 0x55607d8afd63 <unknown>\n#14 0x55607d882c35 <unknown>\n#15 0x55607d8d3138 <unknown>\n#16 0x55607d8d32c7 <unknown>\n#17 0x55607d8e1093 <unknown>\n#18 0x7f20dbb0ab43 <unknown>","statusTrace":"request = <SubRequest 'setup' for <Function test_user_login>>\n\n    @pytest.fixture(scope=\"function\")\n    def setup(request:pytest.FixtureRequest):\n        browser=request.config.getoption(\"--browser\")\n        if(browser in(\"chrome\",\"chrome_headless\")):\n            chrome_options = webdriver.ChromeOptions()\n            chrome_options.set_capability(\n                    \"goog:loggingPrefs\", { \"browser\": \"ALL\"}\n                )\n    \n        match browser:\n            case \"chrome\":\n                driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()),options=chrome_options)\n            case \"chrome_headless\":\n                chrome_options.add_argument(\"headless=new\")\n                driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()),options=chrome_options)\n            case \"firefox\":\n                driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()))\n        driver.get(\"https://sporty.com/news/latest\")\n        # make sure the page is loaded\n        wait= WebDriverWait(driver,30)\n>       wait.until(expected_conditions.visibility_of_element_located(((By.XPATH,\"//*[contains(text(),'Customise your')]\"))))\n\ntests/conftest.py:43: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"0841fd1b65c3b2814e774b1e2af1c8f3\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x7fbc25a9e0e0>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55607d8e8fe3 <unknown>\nE       #1 0x55607d627d36 <unknown>\nE       #2 0x55607d664c4d <unknown>\nE       #3 0x55607d664d61 <unknown>\nE       #4 0x55607d6a06a4 <unknown>\nE       #5 0x55607d6858ed <unknown>\nE       #6 0x55607d69e232 <unknown>\nE       #7 0x55607d685693 <unknown>\nE       #8 0x55607d65803a <unknown>\nE       #9 0x55607d65917e <unknown>\nE       #10 0x55607d8aadbd <unknown>\nE       #11 0x55607d8aec6c <unknown>\nE       #12 0x55607d8b84b0 <unknown>\nE       #13 0x55607d8afd63 <unknown>\nE       #14 0x55607d882c35 <unknown>\nE       #15 0x55607d8d3138 <unknown>\nE       #16 0x55607d8d32c7 <unknown>\nE       #17 0x55607d8e1093 <unknown>\nE       #18 0x7f20dbb0ab43 <unknown>\n\n/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[{"name":"setup","time":{"start":1683293949011,"stop":1683293988467,"duration":39456},"status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55607d8e8fe3 <unknown>\n#1 0x55607d627d36 <unknown>\n#2 0x55607d664c4d <unknown>\n#3 0x55607d664d61 <unknown>\n#4 0x55607d6a06a4 <unknown>\n#5 0x55607d6858ed <unknown>\n#6 0x55607d69e232 <unknown>\n#7 0x55607d685693 <unknown>\n#8 0x55607d65803a <unknown>\n#9 0x55607d65917e <unknown>\n#10 0x55607d8aadbd <unknown>\n#11 0x55607d8aec6c <unknown>\n#12 0x55607d8b84b0 <unknown>\n#13 0x55607d8afd63 <unknown>\n#14 0x55607d882c35 <unknown>\n#15 0x55607d8d3138 <unknown>\n#16 0x55607d8d32c7 <unknown>\n#17 0x55607d8e1093 <unknown>\n#18 0x7f20dbb0ab43 <unknown>\n\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/pluggy/_callers.py\", line 39, in _multicall\n    res = hook_impl.function(*args)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/_pytest/fixtures.py\", line 1130, in pytest_fixture_setup\n    result = call_fixture_func(fixturefunc, request, kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/_pytest/fixtures.py\", line 902, in call_fixture_func\n    fixture_result = next(generator)\n  File \"/home/runner/work/selenium-python/selenium-python/tests/conftest.py\", line 43, in setup\n    wait.until(expected_conditions.visibility_of_element_located(((By.XPATH,\"//*[contains(text(),'Customise your')]\"))))\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/selenium/webdriver/support/wait.py\", line 95, in until\n    raise TimeoutException(message, screen, stacktrace)\n","steps":[],"attachments":[],"parameters":[],"attachmentsCount":0,"shouldDisplayMessage":true,"hasContent":true,"stepsCount":0},{"name":"attach_artifacts_on_failure","time":{"start":1683293949011,"stop":1683293949011,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"attachmentsCount":0,"shouldDisplayMessage":false,"hasContent":false,"stepsCount":0}],"testStage":{"description":"User is able to login","status":"broken","statusMessage":"selenium.common.exceptions.TimeoutException: Message: \nStacktrace:\n#0 0x55607d8e8fe3 <unknown>\n#1 0x55607d627d36 <unknown>\n#2 0x55607d664c4d <unknown>\n#3 0x55607d664d61 <unknown>\n#4 0x55607d6a06a4 <unknown>\n#5 0x55607d6858ed <unknown>\n#6 0x55607d69e232 <unknown>\n#7 0x55607d685693 <unknown>\n#8 0x55607d65803a <unknown>\n#9 0x55607d65917e <unknown>\n#10 0x55607d8aadbd <unknown>\n#11 0x55607d8aec6c <unknown>\n#12 0x55607d8b84b0 <unknown>\n#13 0x55607d8afd63 <unknown>\n#14 0x55607d882c35 <unknown>\n#15 0x55607d8d3138 <unknown>\n#16 0x55607d8d32c7 <unknown>\n#17 0x55607d8e1093 <unknown>\n#18 0x7f20dbb0ab43 <unknown>","statusTrace":"request = <SubRequest 'setup' for <Function test_user_login>>\n\n    @pytest.fixture(scope=\"function\")\n    def setup(request:pytest.FixtureRequest):\n        browser=request.config.getoption(\"--browser\")\n        if(browser in(\"chrome\",\"chrome_headless\")):\n            chrome_options = webdriver.ChromeOptions()\n            chrome_options.set_capability(\n                    \"goog:loggingPrefs\", { \"browser\": \"ALL\"}\n                )\n    \n        match browser:\n            case \"chrome\":\n                driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()),options=chrome_options)\n            case \"chrome_headless\":\n                chrome_options.add_argument(\"headless=new\")\n                driver = webdriver.Chrome(service=ChromeService(ChromeDriverManager().install()),options=chrome_options)\n            case \"firefox\":\n                driver = webdriver.Firefox(service=FirefoxService(GeckoDriverManager().install()))\n        driver.get(\"https://sporty.com/news/latest\")\n        # make sure the page is loaded\n        wait= WebDriverWait(driver,30)\n>       wait.until(expected_conditions.visibility_of_element_located(((By.XPATH,\"//*[contains(text(),'Customise your')]\"))))\n\ntests/conftest.py:43: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"0841fd1b65c3b2814e774b1e2af1c8f3\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x7fbc25a9e0e0>\nmessage = ''\n\n    def until(self, method, message: str = \"\"):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.monotonic() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, \"screen\", None)\n                stacktrace = getattr(exc, \"stacktrace\", None)\n            time.sleep(self._poll)\n            if time.monotonic() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       #0 0x55607d8e8fe3 <unknown>\nE       #1 0x55607d627d36 <unknown>\nE       #2 0x55607d664c4d <unknown>\nE       #3 0x55607d664d61 <unknown>\nE       #4 0x55607d6a06a4 <unknown>\nE       #5 0x55607d6858ed <unknown>\nE       #6 0x55607d69e232 <unknown>\nE       #7 0x55607d685693 <unknown>\nE       #8 0x55607d65803a <unknown>\nE       #9 0x55607d65917e <unknown>\nE       #10 0x55607d8aadbd <unknown>\nE       #11 0x55607d8aec6c <unknown>\nE       #12 0x55607d8b84b0 <unknown>\nE       #13 0x55607d8afd63 <unknown>\nE       #14 0x55607d882c35 <unknown>\nE       #15 0x55607d8d3138 <unknown>\nE       #16 0x55607d8d32c7 <unknown>\nE       #17 0x55607d8e1093 <unknown>\nE       #18 0x7f20dbb0ab43 <unknown>\n\n/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/selenium/webdriver/support/wait.py:95: TimeoutException","steps":[],"attachments":[{"uid":"732cb4dade6373ff","name":"log","source":"732cb4dade6373ff.txt","type":"text/plain","size":1094}],"parameters":[],"attachmentsCount":1,"shouldDisplayMessage":true,"hasContent":true,"stepsCount":0},"afterStages":[{"name":"attach_artifacts_on_failure::0","time":{"start":1683293988524,"stop":1683293988606,"duration":82},"status":"broken","statusMessage":"urllib3.exceptions.MaxRetryError: HTTPConnectionPool(host='localhost', port=46389): Max retries exceeded with url: /session/19550d82327b4e9c34bcedd09c8e170c/se/log (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7fbc25a6afb0>: Failed to establish a new connection: [Errno 111] Connection refused'))\n","statusTrace":"  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/allure_commons/_allure.py\", line 221, in __call__\n    return self._fixture_function(*args, **kwargs)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/_pytest/fixtures.py\", line 918, in _teardown_yield_fixture\n    next(it)\n  File \"/home/runner/work/selenium-python/selenium-python/tests/conftest.py\", line 63, in attach_artifacts_on_failure\n    allure.attach(body=json.dumps(request.cls.driver.get_log(\"browser\"), indent=4), name=\"Console Logs\",attachment_type=allure.attachment_type.JSON)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 1118, in get_log\n    return self.execute(Command.GET_LOG, {\"type\": log_type})[\"value\"]\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py\", line 438, in execute\n    response = self.command_executor.execute(driver_command, params)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py\", line 290, in execute\n    return self._request(command_info[0], url, body=data)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/selenium/webdriver/remote/remote_connection.py\", line 311, in _request\n    response = self._conn.request(method, url, body=body, headers=headers)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/request.py\", line 78, in request\n    return self.request_encode_body(\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/request.py\", line 170, in request_encode_body\n    return self.urlopen(method, url, **extra_kw)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/poolmanager.py\", line 376, in urlopen\n    response = conn.urlopen(method, u.request_uri, **kw)\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/connectionpool.py\", line 815, in urlopen\n    return self.urlopen(\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/connectionpool.py\", line 787, in urlopen\n    retries = retries.increment(\n  File \"/opt/hostedtoolcache/Python/3.10.11/x64/lib/python3.10/site-packages/urllib3/util/retry.py\", line 592, in increment\n    raise MaxRetryError(_pool, url, error or ResponseError(cause))\n","steps":[],"attachments":[],"parameters":[],"attachmentsCount":0,"shouldDisplayMessage":true,"hasContent":true,"stepsCount":0}],"labels":[{"name":"epic","value":"Login functionality"},{"name":"story","value":"Login functionality"},{"name":"tag","value":"@pytest.mark.usefixtures('setup')"},{"name":"parentSuite","value":"tests"},{"name":"suite","value":"test_login"},{"name":"subSuite","value":"TestLogin"},{"name":"host","value":"fv-az548-835"},{"name":"thread","value":"2186-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.test_login"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":true,"retry":true,"extra":{"categories":[],"tags":["@pytest.mark.usefixtures('setup')"]},"source":"ee91158a744e7d39.json","parameterValues":[]}